src/Workspaces/Core/Portable/Workspace/WorkspaceChangeEventArgs.cs

```diff
@@ -21,21 +23,47 @@ public class WorkspaceChangeEventArgs : EventArgs
         public WorkspaceChangeKind Kind { get; }
 
         /// <remarks>
-        /// If linked documents are being changed, there may be multiple events with the same
-        /// <see cref="OldSolution"/> and <see cref="NewSolution"/>.
+        /// If linked documents are being changed, there may be multiple events with the same <see cref="OldSolution"/>
+        /// and <see cref="NewSolution"/>.  Will never be <see langword="null"/>, even for a <see
+        /// cref="WorkspaceChangeKind.SolutionAdded"/> notification.
         /// </remarks>
         public Solution OldSolution { get; }
 
         /// <remarks>
-        /// If linked documents are being changed, there may be multiple events with the same
-        /// <see cref="OldSolution"/> and <see cref="NewSolution"/>.
+        /// If linked documents are being changed, there may be multiple events with the same <see cref="OldSolution"/>
+        /// and <see cref="NewSolution"/>. Will never be <see langword="null"/>, even for a <see
+        /// cref="WorkspaceChangeKind.SolutionRemoved"/> notification.
```

Comment:
>SolutionRemoved [](start = 38, length = 15)

The same here - "when solution is 'removed' it is actually replaced by an empty solution".